schema/schema.go

```diff
@@ -0,0 +1,186 @@
+package schema
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+	"github.com/hashicorp/hcl/hcl/parser"
+	"github.com/hashicorp/hcl/hcl/token"
+)
+
+type Template struct {
+	File      string
+	Resources []*Resource
+}
+
+func Make(files map[string][]byte) ([]*Template, error) {
+	var templates []*Template
+	var err error
+	plains := map[string][]byte{}
+	overrideFiles := []string{}
+	overrides := map[string][]byte{}
+
+	for file := range files {
+		if file == "override.tf" || strings.HasSuffix(file, "_override.tf") {
+			overrideFiles = append(overrideFiles, file)
+		} else {
+			plains[file] = files[file]
+		}
+	}
+	// Override files are loaded last in alphabetical order.
+	sort.Strings(overrideFiles)
+	for _, file := range overrideFiles {
+		overrides[file] = files[file]
+	}
+
+	if templates, err = appendTemplates(templates, plains, false); err != nil {
+		return nil, err
+	}
+	if templates, err = appendTemplates(templates, overrides, true); err != nil {
+		return nil, err
+	}
+
+	return templates, nil
+}
+
+func (t *Template) Find(query ...string) []*Resource {
+	resources := []*Resource{}
+	var provider, providerType, id string
+	for i, attr := range query {
+		switch i {
+		case 0:
+			provider = attr
+		case 1:
+			providerType = attr
+		case 2:
+			id = attr
+		default:
+			// noop
+		}
+	}
+
+	switch provider {
+	case "resource":
+		if providerType != "" {
+			for _, resource := range t.Resources {
+				if id != "" {
+					if resource.Type == providerType && resource.Id == id {
+						resources = append(resources, resource)
+					}
+				} else {
+					if resource.Type == providerType {
+						resources = append(resources, resource)
+					}
+				}
+			}
+		} else {
+			resources = t.Resources
+		}
+		return resources
+	default:
+		return resources
+	}
+}
+
+func appendTemplates(templates []*Template, files map[string][]byte, override bool) ([]*Template, error) {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 17 of function appendTemplates() is high (> 15)

<!-- Comment by [SideCI](https://sideci.com) -->